### PR TITLE
feat(tab-nav-drawer, tabs): Disable animation. #7157

### LINKED ISSUE #7157: [igx-tab] [igx-nav-drawer] Add animations option
## Is your feature request related to a problem? Please describe.
I'd like to have the option to choose whether I want animations on the components `igx-tab` and `igx-nav-drawer`. It'd be something similar to `mat-tab-group` from Angular Material.

I do know their `mat-sidenav-container` does not provide animation options, but if you guys could make it on your `igx-nav-drawer` it would be very nice.

## Describe the solution you'd like
I believe a property similar to Angular Material (`animationDuration="0ms"`) would be nice.

## Describe alternatives you've considered
We can override some styles using CSS, but this alternative looks like an "ugly" workaround.

## Additional context
n/a.

### LINKED ISSUE #8169: Igx-Tab - Transition Theme
Dear Team,

There should be an option to set transition theme as none as couple of corporate customer's doesnt want transition but just a grid or any other information.

Reference Ticket - CAS-208901-R2F7K5

We propose the feature as it is not present urrently.

Please advise.

All the best!

### PR DESCRIPTION
**Pull request by valadzhov**
Closes #7157 
Closes #8169

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts [hunk 1]
```diff
@@ -245,6 +245,9 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
      */
     public visibleItemsWidth: number;
 
+    @Input()
+    public enableAnimation = true;
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/tabs/tabs.component.ts:249]: Should also be `disableAnimation`.

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts (hunk 1 above)
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/tabs/tabs.component.ts]: Changed as suggested.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +17,10 @@ All notable changes for each version of this project will be documented in this
     - **Breaking Change** - Deprecated the `label` property.
 - `igxGridActions`
     - Added `asMenuItems` Input for grid actions - `igx-grid-editing-actions`, `igx-grid-pinning-actions`. When set to true will render the related action buttons as separate menu items with button and label.
-
+- `igxNavigationDrawer`
+    - Added `disableAnimation` input property which enables/disables the animation for the opening and the closing of the drawer. Set to `false` by default.
+- `igxTabs`
+    - Added `disableAnimation` input property which enables/disables the animation for transition of the content of the tabs. Set to `false` by default.
```
Comment (Lipata) [inline CHANGELOG.md:23]:
```suggestion
    - Added `disableAnimation` property which enables/disables the transition animation of the tabs' content. Set to `false` by default.
```

File: projects/igniteui-angular/src/lib/navigation-drawer/README.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ The navigation drawer can either sit above content or be pinned alongside it and
 | `pinThreshold` | number | Minimum device width required for automatic pin to be toggled. Default is 1024, can be set to a falsy value to disable this behavior. |
 | `width` | string| Width of the drawer in its open state. Defaults to "280px".|
 | `miniWidth` | string | Width of the drawer in its mini variant. Defaults to "60px". |
+| `disableAnimation` | boolean | Enables/disables the animation for the  opening and the closing of the drawer. Set to `false` by default.
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/navigation-drawer/README.md:67]:
```suggestion
| `disableAnimation` | boolean | Enables/disables the animation, when toggling the drawer. Set to `false` by default.
```

File: projects/igniteui-angular/src/lib/tabs/README.md [hunk 4]
```diff
@@ -49,6 +49,7 @@ If the total visible items width exceeds the vew port width, scroll buttons are
 | `label` | string | Set the tab item text. |
 | `disabled` | boolean | 	Set if the tab is enabled/disabled.	 |
 | `type` | IgxTabsType | 	Set the tab item sizing mode. By default, `contentfit` is set, the tab item width is sized to its content in the range of min/max width. When the sizing type is `fixed` - all tabs have equal size to fit the view port. |
+| `disableAnimation`| boolean | Enables/disables the animation for transition of the contents of the tabs. Set to `false` by default.
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/tabs/README.md:52]:
```suggestion
| `disableAnimation`| boolean | Enables/disables the transition animation of the tabs' content. Set to `false` by default.
```

File: projects/igniteui-angular/src/lib/navigation-drawer/navigation-drawer.component.ts [hunk 5]
```diff
@@ -204,6 +204,9 @@ export class IgxNavigationDrawerComponent implements
      */
     @Input() public width = '280px';
 
+    @HostBinding ('class.igx-nav-drawer--disable-animation')
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/navigation-drawer/navigation-drawer.component.ts:207]: Add API Code snippet and description

File: projects/igniteui-angular/src/lib/tabs/tabs.component.ts [hunk 6]
```diff
@@ -245,6 +245,9 @@ export class IgxTabsComponent implements IgxTabsBase, AfterViewInit, OnDestroy {
      */
     public visibleItemsWidth: number;
 
+    @Input()
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/tabs/tabs.component.ts:248]: Add API Code snippet and description

**Review by Lipata — CHANGES_REQUESTED** [review]
Add also tests for both properties

File: CHANGELOG.md [hunk 7]
```diff
@@ -17,7 +17,10 @@ All notable changes for each version of this project will be documented in this
     - **Breaking Change** - Deprecated the `label` property.
 - `igxGridActions`
     - Added `asMenuItems` Input for grid actions - `igx-grid-editing-actions`, `igx-grid-pinning-actions`. When set to true will render the related action buttons as separate menu items with button and label.
-
+- `igxNavigationDrawer`
+    - Added `disableAnimation` input property which enables/disables the animation for the opening and the closing of the drawer. Set to `false` by default.
```
[the remark this answers is not in the document]
Comment (valadzhov) [reply CHANGELOG.md:21]: Changed as suggested.

File: CHANGELOG.md (hunk 2 above)
Comment (valadzhov) [reply CHANGELOG.md:23]: Changed as suggested.

File: projects/igniteui-angular/src/lib/navigation-drawer/README.md (hunk 3 above)
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/navigation-drawer/README.md:67]: Changed as suggested.

File: projects/igniteui-angular/src/lib/navigation-drawer/navigation-drawer.component.ts (hunk 5 above)
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/navigation-drawer/navigation-drawer.component.ts:207]: Added API code snippet and description.

File: projects/igniteui-angular/src/lib/tabs/README.md (hunk 4 above)
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/tabs/README.md:52]: Added API code snippet and description.

**Review by desig9stein — DISMISSED** [review]

**Review by Lipata — DISMISSED** [review]

**Review by Lipata — APPROVED** [review]

**Review by Lipata — APPROVED** [review]

**Pull request merged**